### PR TITLE
DataFrameReader.Table

### PR DESCRIPTION
**Pull request by GoEddie**
This add the Table method to DataFrameReader so tables written using DataFrameWriter.SaveAsTable can be read without having to use SparkSession.Sql.

### CODE REVIEW

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs [hunk 1]
```diff
@@ -75,6 +76,11 @@ public void TestSignaturesV2_3_X()
             string textFile = $"{TestEnvironment.ResourceDirectory}people.txt";
             Assert.IsType<DataFrame>(dfr.Text(textFile));
             Assert.IsType<DataFrame>(dfr.Text(textFile, textFile));
+
+            string randomName = $"table_{new Random().Next(100000, 999999).ToString()}";
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs:80]: why does this need to be a random table name ?

File: src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs [hunk 2]
```diff
@@ -255,6 +255,15 @@ public DataFrameReader Options(Dictionary<string, string> options)
         /// <param name="paths">Input paths</param>
         /// <returns>DataFrame object</returns>
         public DataFrame Orc(params string[] paths) => LoadSource("orc", paths);
+        
+        /// <summary>
+        /// Returns the specified table as a DataFrame. This is a table that is configured in the
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs:260]:
```suggestion
        /// Returns the specified table as a DataFrame. This is a table that is configured in the
```

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs (hunk 1 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs:80]: If the files already exist then spark will fail, there used to be a workaround but it was removed with spark 3.0 spark.sql.legacy.allowCreatingManagedTableUsingNonemptyLocation and if the option is set in spark 3.0 it throws an exception.

File: src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs [hunk 3]
```diff
@@ -255,6 +255,15 @@ public DataFrameReader Options(Dictionary<string, string> options)
...
+        /// Hive metabase.
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs:261]: Is this always true?  This only works with tables that are configured in hive ?

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs (hunk 1 above)
Comment (suhsteve) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs:80]: `DataFrameReader.Table(..)` looks like an alias for `SparkSession.Table(..)`.  Can we do the same thing we do for that test?

```csharp
            _spark.Range(10).CreateOrReplaceTempView("testView");
            Assert.IsType<DataFrame>(_spark.Table("testView"));
```

Comment (Niharikadutta) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs:80]: Is this `ToString()` needed? I think this can be simplified.

Comment (GoEddie) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs]: I've copied the test from Spark.Table

Comment (GoEddie) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs]: I've copied it now

File: src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs (hunk 3 above)
Comment (GoEddie) [reply src/csharp/Microsoft.Spark/Sql/DataFrameReader.cs]: I've removed it now

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs:5]: is this still needed ?

Comment (GoEddie) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs]: gone now!

**Review by suhsteve — APPROVED** [review]
LGTM.  Thanks @GoEddie !

**Comment by Niharikadutta** [comment]
LGTM. Thanks @GoEddie !

**Review by imback82 — APPROVED** [review]
LGTM, thanks @GoEddie!

**Pull request merged**